module/FinnaApi/src/FinnaApi/Controller/ListApiController.php

```diff
@@ -0,0 +1,166 @@
+        $this->recordFormatter = $rf;
+        foreach ($rf->getRecordFields() as $fieldName => $fieldSpec) {
+            if (!empty($fieldSpec['vufind.default'])) {
+                $this->defaultRecordFields[] = $fieldName;
+            }
+        }
+    }
+
+    /**
+     * Get Swagger specification JSON fragment for services provided by the
+     * controller
+     *
+     * @return string
+     */
+    public function getSwaggerSpecFragment()
+    {
+        // TODO: Implement getSwaggerSpecFragment() method.
+        return '';
+    }
+
+    /**
+     * List action
+     *
+     * @return \Laminas\Http\Response
+     */
+    public function listAction()
+    {
+        $requestParams = $this->getRequest()->getQuery()->toArray()
+            + $this->getRequest()->getPost()->toArray();
+
+        $id = $requestParams['id'] ?? false;
+        if (!$id) {
+            return $this->output([], self::STATUS_ERROR, 400, 'Missing id');
+        }
+
+        try {
+            $list = $this->getTable('UserList')->getExisting($id);
+            if (!$list->isPublic()) {
+                return $this->output(
+                    [], self::STATUS_ERROR, 403, 'Permission denied'
+                );
+            }
+        } catch (RecordMissingException $e) {
+            return $this->output([], self::STATUS_ERROR, 404, 'List not found');
+        }
+
+        try {
+            $results = $this->serviceLocator
+                ->get(\VuFind\Search\Results\PluginManager::class)->get('Favorites');
+            $params = $results->getParams();
+
+            $params->initFromRequest(new Parameters($requestParams));
+
+            $results->performAndProcessSearch();
+            $listObj = $results->getListObject();
+
+            $response = [
+                'list' => [
+                    'id' => $listObj->id,
+                    'title' => $listObj->title,
+                    'description' => $listObj->description
+                    // TODO: Add all list data to response
+                ]
+            ];
+
+            if ($this->listTagsEnabled()) {
+                $listTags = $this->getTable('Tags')
+                    ->getForList($listObj->id, $listObj->user_id);
+                // TODO: Add list tags to response
+            }
+
+            $response['resultCount'] = $results->getResultTotal();
+
+            $records = $this->recordFormatter->format(
+                $results->getResults(), $this->defaultRecordFields
+            );
+            if ($records) {
+                $response['records'] = $records;
+                $response['notes'] = [];
+                foreach ($results->getResults() as $result) {
+                    $response['notes'][] = $result->getListNotes($listObj->id);
+                }
+            }
+
+            return $this->output($response, self::STATUS_OK);
+        } catch (ListPermissionException $e) {
+            return $this->output([], self::STATUS_ERROR, 403, 'Permission denied');
```

Comment:
Ennemmin 404 'List not found' tässäkin. Ei tartte kysyjälle kertoa, onko listaa olemassa.